lib/src/signed_video_h26x_common.c

```diff
@@ -845,17 +845,20 @@ update_gop_hash(void *crypto_handle, gop_info_t *gop_info)
 }
 
 /* compute_partial_gop_hash()
- * Takes all the hash nalus from |hash_list| and hash it.
+ * Takes all the NALU hashes from |hash_list| and hash it.
  */
 svrc_t
 compute_partial_gop_hash(signed_video_t *self)
 {
   gop_info_t *gop_info = self->gop_info;
   uint8_t *hash = gop_info->computed_gop_hash;
-  if (gop_info->list_idx <= 0) {
-    // The list index is out of bounds (either zero or negative).
-    // TODO: Handle cases where list_idx is negative by returning SV_INVALID_PARAMETER for invalid
-    // list index.
+  if (gop_info->list_idx < 0) {
+    // TODO: When list_idx is greater than 0, it means hash_list run out of memory and partial
+    // GOP hash can't be computed. In this case,instead of DOCUMENT_HASH, GOP_HASH should be used.
```

Comment:
Not really correct. Write something like

TODO: When list_idx < 0 it means that there was not enough memory allocated for the hash_list to add another hash to it. The effect is that Signed Video will run with a GOP level authenticity. The current implementation of this new gop_hash cannot handle such a fallback since it is computed from the hash_list (which currently is in a bad state). This has to be reworked.